File: array/basics/0485_max_consecutive_ones.js
/*
Given a binary array nums, return the maximum number of consecutive 1's 
in the array.
*/

// http://leetcode.com/problems/max-consecutive-ones/

// traverse
// time compelxity O(n)
// space complexity O(1)
var findMaxConsecutiveOnes = function (nums) {
  let count = 0;
  let maxCount = 0;
  for (let i = 0; i < nums.length; i++) {
    if (nums[i] === 1) {
      count++;
    } else {
      maxCount = Math.max(count, maxCount);
      count = 0;
    }
  }
  return Math.max(count, maxCount);
};

File: array/basics/0485_max_consecutive_ones.py
# Given a binary array nums, return the maximum number of consecutive 1's 
# in the array.

# http://leetcode.com/problems/max-consecutive-ones/

# binary array featuers
class Solution:
    def findMaxConsecutiveOnes(self, nums: list[int]) -> int:
        max_cnt, cnt = 0, 0
        for i in nums:
            if i == 1:
                cnt += 1
                max_cnt = max(max_cnt, cnt)
            else:
                cnt = 0
        return max_cnt
            